Public/Sdk/SelfHost/BuildXL/BuildXLSdk.dsc

```diff
@@ -317,6 +324,29 @@ export function test(args: TestArguments) : TestResult {
             /* limitCategories:      */ args.runTestArgs && args.runTestArgs.limitGroups,
             /* skipCategories:       */ args.runTestArgs && args.runTestArgs.skipGroups,
             /* untrackTestDirectory: */ args.runTestArgs && args.runTestArgs.untrackTestDirectory);
+
+        if (Flags.buildRequiredAdminPrivilegeTestInVm) {
+            // QTest doesn't really work when the limit categories filter out all the tests.
+            // Basically, the logic below follows standalone test runner.
+            const untrackedFramework = importFrom("Sdk.Managed.Testing.XUnit.UnsafeUnDetoured").framework;
+            const trackedFramework = importFrom("Sdk.Managed.Testing.XUnit").framework;
+            const untracked = args.testFramework && args.testFramework.name.endsWith(untrackedFramework.name);
+            const framework = untracked ? untrackedFramework : trackedFramework;

```

Comment:
Why this complicated logic,
can you just test if the framework === the qtest framework one?

Reply:
Actually why didn't you just create an admin test framework?

---
In reply to: [296363478](https://github.com/microsoft/BuildXL/pull/492#discussion_r296363478) [](ancestors = 296363478)

Reply:
QTest doesn't work. QTest framework wraps the other frameworks and I need to get the wrapped frameworks. This is what I did for the standalone test.

Sure we can create an admin test framework, but I think it's overkill.

---
In reply to: [296363670](https://github.com/microsoft/BuildXL/pull/492#discussion_r296363670) [](ancestors = 296363670,296363478)